src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs

```diff
@@ -25,8 +25,11 @@ public static bool TryGetValue<TItem>(this IMemoryCache cache, object key, out T
         {
             if (cache.TryGetValue(key, out object result))
             {
-                value = (TItem)result;
-                return true;
+                if (result is TItem item)
```

Comment:
This has some subtle side-effects. The behavior of `Get<TItem>` changes, but not `Get` or `GetOrCreate<TItem>` or `GetOrCreateAsync<TItem>`. Change `Get<TItem>` to throw?

Reply:
oh, you are right. What do you suggest?

Reply:
Change `Get<TItem>` to call `Get` and Cast instead of calling `TryGet<TItem>`.